task_processing/plugins/kubernetes/kubernetes_pod_executor.py

```diff
@@ -420,6 +451,37 @@ def _pending_event_processing_loop(self) -> None:
 
         logger.debug("Exiting Pod event processing - stop requested.")
 
+    def _reconcile_task_loop(self) -> None:
+        """
+        Run in a thread to reconcile task_metadata from k8s.
+        """
+        logger.info(f'Waiting {self.refresh_reconciliation_thread_grace}s before doing work')
+        sleep(self.refresh_reconciliation_thread_grace)
+        logger.debug("Starting Pod task config reconciliation.")
+        while not self.stopping:
+            try:
+                # fetch all pods in the target namespace in one request so that we
+                # don't block on making N serial requests to the Kubernetes API
+                pods = self.kube_client.get_pods(namespace=self.namespace)
+            except Exception:
+                logger.exception(
+                    f"Hit an exception attempting to fetch pods in namespace {self.namespace}")
+                pods = None
+                continue
+
+            if pods is not None:
+                # we've previously bulk-fetched all the pods running in the target
+                # namespace - we'll now filter these by the tasks down to only
+                # those we know about so that we only reconcile the state for what
+                # we actually need and not any other cruft that may exist.
+                task_configs_pods = self._filter_task_configs_in_pods(pods)
+                if task_configs_pods:
+                    for task_metadata, pod in task_configs_pods:
```

Comment:
Since _filter_task_configs_in_pods will only return tasks with pods found from k8s api, this means we'll never actually reconcile tasks we know about but *don't* have pods.  That seems like we might still have the edge case where we can miss pods going away and never being reconciled by tron/setting them to `LOST`.  

Perhaps we should iterate over all task_metadata instead of only those with discovered pods.

Reply:
good catch! I'll update by iterating over task_metadata instead :)

Reply:
jfong++ - i totally tunnel-visioned/brain-fog'd over this case!